src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -149,6 +149,12 @@ class OrderedRVD private(
     OrderedRVD(typ, orderedPartitioner, rdd)
   }
 
+  def blockCoalesce(partitionEnds: Array[Int]): OrderedRVD = {
+    assert(partitionEnds.last == partitioner.numPartitions - 1)
+    assert(partitionEnds.zip(partitionEnds.tail).forall { case (i, inext) => i < inext })
```

Comment:
I'm having off-by-one head pains here.  What's the invariant on partitionEnds?

partitionEnds is the length of the new number of partitions, and they go from [0, pE(0)], ... [pE(n - 2), pE(n - 1)]?  So partitionEnds(0) >= 0?

So we need assert(partitionEnds(0) >= 0)?

Reply:
yeah. I'll add.